Sources/Purchasing/StoreKit2/StoreKit2StorefrontListener.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreKit2StorefrontListener.swift
+//
+//  Created by Juanpe Catalán on 5/5/22.
+
+import Foundation
+import StoreKit
+
+protocol StoreKit2StorefrontListenerDelegate: AnyObject {
+    func storefrontDidUpdate()
+}
+
+@available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+class StoreKit2StorefrontListener {
+
+    private(set) var taskHandle: Task<Void, Never>? {
+        didSet {
+            if self.taskHandle != oldValue {
+                oldValue?.cancel()
+            }
+        }
+    }
+
+    weak var delegate: StoreKit2StorefrontListenerDelegate?
+
+    init(delegate: StoreKit2StorefrontListenerDelegate?) {
+        self.delegate = delegate
+    }
+
+    func listenForStorefrontChanges() {
+        taskHandle = Task { [weak self] in
+            for await _ in StoreKit.Storefront.updates {
+                guard let self = self else { break }
+                self.delegate?.storefrontDidUpdate()
+            }
+        }
+    }
+
+    deinit {
+        taskHandle?.cancel()
+        taskHandle = nil
```

Comment:
This is duplicated, you can just do `self.taskHandle = nil` now :)
The advantage of what I suggested is that now it's impossible to have a reference to a cancelled task :)

Reply:
Yeah! good catch! I forgot to update this line 🤦🏼‍♂️

Reply:
Done! ✅  ([171db44](https://github.com/RevenueCat/purchases-ios/pull/1557/commits/171db446d23bf8e2adfbf138cb65419f16333c07))

Reply:
I've reverted the change because when the listener is released, the `didSet` observer is not called 😅 so the task is not cancelled. [I've cancelled the task when the listener is released](https://github.com/RevenueCat/purchases-ios/blob/ad5e799ac4cf64a1e43076ab3b98b570f58c2e16/Sources/Purchasing/StoreKit2/StoreKit2StorefrontListener.swift#L52) and [modified the test cases](https://github.com/RevenueCat/purchases-ios/blob/ad5e799ac4cf64a1e43076ab3b98b570f58c2e16/Tests/StoreKitUnitTests/StoreKit2StorefrontListenerTests.swift#L29) to cover this scenario.

Reply:
Oooh that's a great catch I forgot!!